File: anuglar-ui/angular-ecommerce/src/app/common/purchase.ts
import {Address} from "./address";
import {Order} from "./order";
import {OrderItem} from "./order-item";
import {Customer} from "./customer";

//Customer
export class Purchase {
  customer!: Customer;
  shippingAddress!: Address;
  billingAddress!: Address;
  order!: Order;
  orderItems!: OrderItem[];
}

File: src/main/java/learn/luv2code/ecommerce/dto/Purchase.java
package learn.luv2code.ecommerce.dto;

import learn.luv2code.ecommerce.entity.Address;
import learn.luv2code.ecommerce.entity.Customer;
import learn.luv2code.ecommerce.entity.Order;
import learn.luv2code.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}
